tests/integration/deploy/test_deploy_command_cdk.py

```diff
@@ -0,0 +1,742 @@
+import json
+import os
+import shutil
+import tempfile
+import uuid
+import time
+import copy
+import logging
+from unittest import skipIf
+
+import boto3
+import docker
+from parameterized import parameterized
+
+from samcli.lib.config.samconfig import DEFAULT_CONFIG_FILE_NAME
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from tests.integration.deploy.deploy_integ_base import DeployIntegBase
+from tests.integration.package.package_integ_base import CdkPackageIntegPythonBase, PackageIntegBase
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import CommandResult, run_command, run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_DEPLOY_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_SLEEP = 3
+TIMEOUT = 300
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+LOG = logging.getLogger(__name__)
+
+
+@skipIf(SKIP_DEPLOY_TESTS, "Skip deploy tests in CI/CD only")
+class TestDeployCdkPython(CdkPackageIntegPythonBase, DeployIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.docker_client = docker.from_env()
+        cls.local_images = [
+            ("alpine", "latest"),
+        ]
+        # setup some images locally by pulling them.
+        for repo, tag in cls.local_images:
+            cls.docker_client.api.pull(repository=repo, tag=tag)
+        # setup signing profile arn & name
+        cls.signing_profile_name = os.environ.get("AWS_SIGNING_PROFILE_NAME")
+        cls.signing_profile_version_arn = os.environ.get("AWS_SIGNING_PROFILE_VERSION_ARN")
+
+        CdkPackageIntegPythonBase.setUpClass()
+        DeployIntegBase.setUpClass()
+
+    def setUp(self):
+        self.cf_client = boto3.client("cloudformation")
+        self.sns_arn = os.environ.get("AWS_SNS")
+        self.stack_names = []
+        time.sleep(CFN_SLEEP)
+        super().setUp()
+
+    def tearDown(self):
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "build"), ignore_errors=True)
+        for stack_name in self.stack_names:
+            # because of the termination protection, do not delete aws-sam-cli-managed-default stack
+            if stack_name != SAM_CLI_STACK_NAME:
+                self.cf_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    def _method_to_stack_name(self, method_name):
+        """Method expects method name which can be a full path. Eg: test.integration.test_deploy_command.method_name"""
+        method_name = method_name.split(".")[-1]
+        return f"{method_name.replace('_', '-')}-{CFN_PYTHON_VERSION_SUFFIX}"
+
+    def _replace_cloud_assembly_stack_name(self, manifest_path, original_stack_name, new_stack_name):
```

Comment:
Question: What's the purpose of this function/is it used in these tests?

Reply:
sorry, it's not needed anymore. Will remove it.